apps/server/src/modules/post/services/post.service.ts

```diff
@@ -269,4 +257,104 @@ export class PostService {
 
 		return true;
 	}
+
+	async getUserPostsWithDetails({ query, meta }: GetUserPostsProps) {
+    const userId = meta.userId;
+
+    const user = await this.userModel.findOne({ _id: userId });
```

Comment:
Tu consegue puxar o accountIds de cada usuário direto do findOne. Testa antes para ver se funciona

```suggestion
    const user = await this.userModel.findOne(
    { _id: userId },
    {
       _id: 0,
       InstagramAccounts: 1
     }
    ).then((accounts) => account.accountId );
```